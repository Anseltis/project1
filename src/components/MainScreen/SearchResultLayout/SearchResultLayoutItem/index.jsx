import React from 'react'
import { Link } from 'react-router-dom'

export const SearchResultLayoutItem = ({item, onclick, dataKey}) => {
    const clickHandler  = () => onclick({ dataKey })
    return (
        <Link to={`/${item.id}`}>
        <div className='film-layout__item' onClick={clickHandler}>
        <img className='film-layout__cover' src={item.poster_path} alt=''/>
        <div className='film-layout__title-container'>
            <span className='film-layout__title'>{item.title}</span>
            <span className='film-layout__year'>{item.release_date}</span>
        </div>
         <span className='film-layout__genre'>{item.genres[0]}</span>
    </div>
            </Link>)
}
