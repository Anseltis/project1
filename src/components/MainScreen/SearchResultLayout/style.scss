%film-layout-child-shared {
  font-family: Arial, sans-serif;
  font-size: .75rem;
  color: #808080;
}

.film-layout {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  flex-direction: row;
  align-items: flex-start;
  list-style: none;
  margin: 0;
  padding: 0 3.125rem;

  &__cover {
    width: 15.63rem;
  }

  &__title {

    font-family: Arial, sans-serif;
    font-size: 1rem;
    font-weight: 600;
    text-transform: uppercase;
    margin-left: .625rem;
    max-width: 60%;

  }

  &__link {
    text-decoration: none;
    color: inherit;
  }

  &__title-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    flex-direction: row;
    align-items: flex-start;

    margin-top: .3125rem;
    width: 15.63rem;
  }

  &__year {
    @extend %film-layout-child-shared;
    display: block;
    border: .0625rem solid #808080;
    margin-right: .625rem;
    padding: .0625rem .3125rem;
  }

  &__genre {
    @extend %film-layout-child-shared;
    margin-left: .625rem;
  }

  &__item {
    margin-bottom: 3.125rem;
    padding: 0 3.125rem;
    position: relative;
    width: 16.66rem;
  }
}
