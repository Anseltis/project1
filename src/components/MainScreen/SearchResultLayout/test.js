import React from 'react'
import renderer from 'react-test-renderer'

import { SearchResultLayout } from 'components/MainScreen/SearchResultLayout/index'

const filmsMock = [
  {
    id: '',
    genres: ['']
  }
]

test('SearchResultLayout', () => {
  const wrapper = renderer
    .create(<SearchResultLayout films={filmsMock} />)
    .toJSON()
  expect(wrapper).toMatchSnapshot()
})
