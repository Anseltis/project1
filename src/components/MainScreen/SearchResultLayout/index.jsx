import React from 'react'
// import PropTypes from 'prop-types'

import { SearchResultLayoutItem } from 'components/MainScreen/SearchResultLayout/SearchResultLayoutItem'

import './style.scss'

export const SearchResultLayout = ({films, onclick}) => {
    const filmsList = films.map((film, index) =>
        <SearchResultLayoutItem item={film} key={film.id} onclick={onclick} dataKey={index}/>
    )
    return (<ul className='film-layout'>{filmsList}</ul>)
}

