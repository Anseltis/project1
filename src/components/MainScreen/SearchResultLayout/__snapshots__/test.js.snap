// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultLayout 1`] = `
<ul
  className="film-layout"
>
  <div
    className="film-layout__item"
    onClick={[Function]}
  >
    <img
      alt=""
      className="film-layout__cover"
    />
    <div
      className="film-layout__title-container"
    >
      <span
        className="film-layout__title"
      />
      <span
        className="film-layout__year"
      />
    </div>
    <span
      className="film-layout__genre"
    >
      
    </span>
  </div>
</ul>
`;
